### PR TITLE
Add timeout parameter

### PR DESCRIPTION
**Pull request by mauriciovasquezbernal**
- Add a timeout parameter that specifies the number of seconds that a gadget will run
- Add some logic to handle it in filetop

### CODE REVIEW

**Review by eiffel-fl — APPROVED** [review]
I tested it and it works as expected, thank you for it!

Nonetheless, I have several remarks:
1. We should really add a guide under `/docs` which would describe the `CommonFlags` (it can be done later, for example once #500 is merged).
2. Can you please add this:

```
--timeout int            Number of seconds that the gadget will run for
```

to output of `kubectl gadget --help` in `README.md`?

3. We should also add the `filetop` particularly to `tcptop` and `biotop` (can be done later).

**Comment by mauriciovasquezbernal** [comment]
>     1. We should really add a guide under `/docs` which would describe the `CommonFlags` (it can be done later, for example once #500 is merged).
> 

I agree! 

>     2. Can you please add this:
> 
> 
> ```
> --timeout int            Number of seconds that the gadget will run for
> ```
> 
> to output of `kubectl gadget --help` in `README.md`?
> 

I'm not sure where do you want to add it. We don't have any flag in `README.md`.

>     3. We should also add the `filetop` particularly to `tcptop` and `biotop` (can be done later).

What do you mean? Add the help output of filetop to a readme?

**Comment by eiffel-fl** [comment]
> > ```
> > 1. We should really add a guide under `/docs` which would describe the `CommonFlags` (it can be done later, for example once #500 is merged).
> > ```
> 
> I agree!
> 
> > ```
> > 2. Can you please add this:
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ```
> > --timeout int            Number of seconds that the gadget will run for
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > to output of `kubectl gadget --help` in `README.md`?
> 
> I'm not sure where do you want to add it. We don't have any flag in `README.md`.

Oh sorry, I thought we have a description of them in the README, then nevermind.

> > ```
> > 3. We should also add the `filetop` particularly to `tcptop` and `biotop` (can be done later).
> > ```
> 
> What do you mean? Add the help output of filetop to a readme?

No, to port the work you did on `filetop` to `tcptop` and `biotop`.

**Comment by mauriciovasquezbernal** [comment]
> No, to port the work you did on filetop to tcptop and biotop.

Right. I updated this PR to handle it in tcptop. Since https://github.com/kinvolk/inspektor-gadget/pull/551 is not merged yet, I didn't touch biotop at all.

**Comment by eiffel-fl** [comment]
> Right. I updated this PR to handle it in tcptop. Since https://github.com/kinvolk/inspektor-gadget/pull/551 is not merged yet, I didn't touch biotop at all.

Thank you for `tcptop` but I did not mean you had to do it (I could have handle it in a separate PR).
I tested it and it works fine.
Regarding `biotop`, I will either address this in another PR, or rebase once yours is merged.

File: cmd/kubectl-gadget/networkpolicyadvisor.go
Comment (eiffel-fl) [inline cmd/kubectl-gadget/networkpolicyadvisor.go:66]: Too brutal `s///`?
Also, I am concerned this is a problem but I would like to first have #500 merged before doing it.
Particularly because I am sure we can group some variables per gadget type.

**Review by eiffel-fl — COMMENTED** [review]
Please see my comment before pushing the merge button:

File: cmd/kubectl-gadget/networkpolicyadvisor.go
Comment (mauriciovasquezbernal) [reply cmd/kubectl-gadget/networkpolicyadvisor.go]: > Too brutal s///?

ops, replaced too many things. 

> Also, I am concerned this is a problem but I would like to first have https://github.com/kinvolk/inspektor-gadget/pull/500 merged before doing it.
Particularly because I am sure we can group some variables per gadget type.

makes sense to me.

**Pull request merged**
